swan-lake/development-tutorials/observability/supported-observability-tools-and-platforms/prometheus.md

```diff
@@ -0,0 +1,164 @@
+    
+    scrape_configs:
+      - job_name: 'prometheus'
+        static_configs:
+          - targets: ['a.b.c.d:9797']
+    ```
+
+    Here, the `'a.b.c.d:9797'` targets should contain the host and port of the `/metrics` service that is exposed from 
+    Ballerina for metrics collection. Add the IP of the host in which the Ballerina service is running as `a.b.c.d` and its
+    port (default `9797`).
+    If you need more information, go to the <a href="https://prometheus.io/docs/introduction/first_steps/" target="_blank">Prometheus documentation</a>.
+    
+    If your Ballerina service is running on localhost and Prometheus in a Docker container,
+    add the target as `host.docker.internal:9797` to access the localhost from Docker.
+
+3.  Start the Prometheus server in a Docker container with the command below.
+
+    ```
+    $ docker run -p 19090:9090 -v <path_to_prometheus.yml>:/etc/prometheus/ prom/prometheus
+    ```
+
+## Step 2 - Import Ballerina Prometheus extension
+
+To include the Prometheus extension into the executable, the `ballerinax/prometheus` module needs to be imported into your Ballerina project `main.bal` file.
+
+```ballerina
+import ballerinax/prometheus as _;
+```
+
+To support Prometheus as the metrics reporter, an HTTP endpoint starts with the context of `/metrics` in default port `9797` when starting the Ballerina service.
+
+## Step 3 - Configure Ballerina runtime configurations
+You can set up prometheus for your Ballerina project using configurations similar to the following in your `Config.toml` file.
+
+```toml
+[ballerina.observe]
+metricsEnabled=true
+metricsReporter="prometheus"
+
+[ballerinax.prometheus]
+port=9797
+host="0.0.0.0"
+```
+
+Configuration key | Description | Default value | Possible values 
+--- | --- | --- | --- 
+`ballerinax.prometheus.port` | The value of the port to which the '/metrics' service will bind. This service will be used by Prometheus to scrape the information of the Ballerina service. | `9797` | Any suitable value for port 0 - 0 - 65535. However, within that range, ports 0 - 1023 are generally reserved for specific purposes, therefore it is advisable to select a port without that range. 
+`ballerinax.prometheus.host` | The name of the host to which the '/metrics' service will bind. This service will be used by Prometheus to scrape the information of the Ballerina service. | `0.0.0.0` | IP or Hostname or 0.0.0.0 of the node in which the Ballerina service is running.
+
+## Step 4 - Run the Ballerina service
+
+When Ballerina observability is enabled, the Ballerina runtime exposes internal metrics via an HTTP endpoint (`/metrics`) for metrics monitoring and the metrics will be published to Prometheus. Prometheus should be configured to scrape metrics from the metrics HTTP endpoint in Ballerina.
+
+Run the following command to start the Ballerina service.
+
+```
+$ bal run
+
+Compiling source
+
+Running executable
+
+ballerina: started Prometheus HTTP listener 0.0.0.0:9797
```

Comment:
In ballerina 12.3 version I dont see the last line. Please check

<img width="1184" alt="image" src="https://github.com/user-attachments/assets/7b117586-960a-4502-b322-237565e1b38b" />